src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -1,15 +1,246 @@
+
+  module Graphql = struct
+    let port = 3085
+
+    (* queries on localhost because of port forwarding *)
+    let uri =
+      Uri.make
+        ~host:Unix.Inet_addr.(localhost |> to_string)
+        ~port:3085 ~path:"graphql" ()
+
+    let set_port_forwarding ~logger ~name ~namespace =
+      let args =
+        ["port-forward"; name; "--namespace"; namespace; string_of_int port]
+      in
+      [%log info] "Running kubectl %s\n" String.(concat args ~sep:" ") ;
+      let%bind.Deferred.Or_error.Let_syntax proc =
+        Process.create ~prog:"kubectl" ~args ()
+      in
+      Exit_handlers.register_handler ~logger
+        ~description:"Kubectl port forwarder" (fun () ->
+          ignore Signal.(send kill (`Pid (Process.pid proc))) ) ;
+      Process.collect_stdout_and_wait proc
+
+    module Client = Graphql_lib.Client.Make (struct
+      let preprocess_variables_string = Fn.id
+
+      let headers = String.Map.empty
+    end)
+
+    module Unlock_account =
+    [%graphql
+    {|
+          mutation ($password: String!,
+          $public_key: PublicKey!) {
+             unlockAccount(input: {password: $password, publicKey: $public_key }) {
+                 public_key: publicKey @bsDecoder(fn: "Decoders.public_key")
+             }
+          }
+    |}]
+
+    module Send_payment =
+    [%graphql
+    {|
+          mutation ($sender: PublicKey!,
+          $receiver: PublicKey!,
+          $amount: UInt64!,
+          $token: UInt64,
+          $fee: UInt64!,
+          $nonce: UInt32,
+          $memo: String) {
+          sendPayment(input:
+            {from: $sender, to: $receiver, amount: $amount, token: $token, fee: $fee, nonce: $nonce, memo: $memo}) {
+              payment {
+        id
+      }
+    }
+  }
+  |}]
+  end
+
+  let send_payment ~logger ~namespace ~node ~sender ~receiver ~amount ~fee =
+    [%log info] "Running send_payment test"
+      ~metadata:[("namespace", `String namespace); ("node", `String node)] ;
+    let open Deferred.Or_error.Let_syntax in
+    let%bind () = set_name_from_network ~logger ~namespace ~node in
+    let name = Option.value_exn (get_full_name ()) in
+    Deferred.don't_wait_for
+      ( match%map.Deferred.Let_syntax
+          Graphql.set_port_forwarding ~logger ~name ~namespace
+        with
+      | Ok _ ->
+          (* not reachable, port forwarder does not terminate *)
+          ()
+      | Error err ->
+          [%log fatal] "Error running k8s port forwarding"
+            ~metadata:[("error", `String (Error.to_string_hum err))] ;
+          failwith "Could not run k8s port forwarding" ) ;
+    let unlock_sender_account_graphql () =
+      let num_tries = 10 in
+      let initial_delay_sec = 30.0 in
+      let retry_delay_sec = 30.0 in
+      let unlock_account_obj =
+        Graphql.Unlock_account.make ~password:"naughty blue worm"
+          ~public_key:(Graphql_lib.Encoders.public_key sender)
+          ()
+      in
+      (* GraphQL not immediately available, retry as needed *)
```

Comment:
I would think you wouldn't need to handle this if you are waiting for the GraphQL initialization in the logs.

Reply:
You'd think ... but GraphQL queries failed. I think that the wait for `init` will make this delay unneeded.

Reply:
Nope, still get "Connection refused" sometimes on first try after passing `wait_for_init` gate.

Reply:
Hmm, how unfortunate. Maybe we should make an issue to fix this at some point (cc @bkase @figitaki).